tests/PySys/c8y_restart_bridge/run.py

```diff
@@ -0,0 +1,41 @@
+import sys
+
+sys.path.append("environments")
+from environment_c8y import EnvironmentC8y
+
+"""
+Bridge restart
+
+Given a configured system with configured certificate
+When we setup EnvironmentC8y
+When we validate EnvironmentC8y
+When we cleanup EnvironmentC8y
+Then then the test has passed
+
+So far just use the EnvironmentC8y nothing else:
+see ../environment/environment_c8y
+"""
+
+
+class RestartBridge(EnvironmentC8y):
+    def setup(self):
+        super().setup()
+        self.log.info("Setup")
+        self.addCleanupFunction(self.mycleanup)
```

Comment:
I think this would lead to double registration of the `cleanup` routine as the parent class is already registering the cleanup function in its `setup` method.

Reply:
Very good finding. It actually registered the same cleanup function twice. I removed the call to the parent method and renamed the cleanup functions in the environments. Even though PySys filters it out in the addCleanupFunction it wasn't the best design :)